pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -422,14 +500,35 @@ func (c *Cluster) provisionPodTemplateSpec(devices []rookalpha.Device, selection
 
 	// ceph-volume --dmcrypt uses cryptsetup that synchronizes with udev on
 	// host through semaphore
-	podSpec.HostIPC = storeConfig.EncryptedDevice
+	podSpec.HostIPC = osdProps.storeConfig.EncryptedDevice
 
 	return &v1.PodTemplateSpec{
 		ObjectMeta: podMeta,
 		Spec:       podSpec,
 	}, nil
 }
 
+func (c *Cluster) getPVCInitContainer(pvc v1.PersistentVolumeClaimVolumeSource) v1.Container {
```

Comment:
please add comments to this method to explain why we need this container to workaround the block device issues.